table/manifest_operations_test.go

```diff
@@ -0,0 +1,514 @@
+			{"id": 2, "data": "bar", "value": 200},
+			{"id": 3, "data": "baz", "value": 300}
+		]`,
+	})
+	s.Require().NoError(err)
+}
+
+func (s *ManifestOperationsTestSuite) SetupTest() {
+	s.location = filepath.ToSlash(strings.Replace(s.T().TempDir(), "#", "", -1))
+}
+
+func (s *ManifestOperationsTestSuite) TearDownSuite() {
+	s.arrTable.Release()
+}
+
+func (s *ManifestOperationsTestSuite) getMetadataLoc() string {
+	return fmt.Sprintf("%s/metadata/%05d-%s.metadata.json",
+		s.location, 1, uuid.New().String())
+}
+
+func (s *ManifestOperationsTestSuite) writeParquet(fio iceio.WriteFileIO, filePath string, arrTbl arrow.Table) {
+	fo, err := fio.Create(filePath)
+	s.Require().NoError(err)
+
+	s.Require().NoError(pqarrow.WriteTable(arrTbl, fo, arrTbl.NumRows(),
+		nil, pqarrow.DefaultWriterProps()))
+}
+
+func (s *ManifestOperationsTestSuite) createTable(identifier table.Identifier, formatVersion int, spec iceberg.PartitionSpec, props iceberg.Properties) *table.Table {
+	meta, err := table.NewMetadata(s.tableSchema, &spec, table.UnsortedSortOrder,
+		s.location, props)
+	s.Require().NoError(err)
+
+	return table.New(
+		identifier,
+		meta,
+		s.getMetadataLoc(),
+		func(ctx context.Context) (iceio.IO, error) {
+			return iceio.LocalFS{}, nil
+		},
+		&mockedCatalog{},
+	)
+}
+
+func (s *ManifestOperationsTestSuite) getFS(tbl *table.Table) iceio.WriteFileIO {
+	fs, err := tbl.FS(s.ctx)
+	s.Require().NoError(err)
+
+	return fs.(iceio.WriteFileIO)
+}
+
+func (s *ManifestOperationsTestSuite) createTableWithManyFiles(identifier table.Identifier, numFiles int, enableMerge bool) (*table.Table, []string) {
+	// Configure table properties for manifest merging
+	props := iceberg.Properties{"format-version": "2"}
+	if enableMerge {
+		// Enable manifest merging with aggressive settings for testing
+		props[table.ManifestMergeEnabledKey] = "true"
+		props[table.ManifestMinMergeCountKey] = "2"      // Merge when we have 2+ manifests
+		props[table.ManifestTargetSizeBytesKey] = "4096" // Small target size for testing
+	} else {
+		props[table.ManifestMergeEnabledKey] = "false"
+	}
+
+	tbl := s.createTable(identifier, 2, *iceberg.UnpartitionedSpec, props)
+
+	files := make([]string, 0, numFiles)
+	fs := s.getFS(tbl)
+
+	// Create and add files one by one to generate multiple manifest files
+	for i := 0; i < numFiles; i++ {
+		filePath := fmt.Sprintf("%s/data/test-%d.parquet", s.location, i)
+		s.writeParquet(fs, filePath, s.arrTable)
+		files = append(files, filePath)
+
+		// Add each file in a separate transaction to create multiple manifests
+		tx := tbl.NewTransaction()
+		s.Require().NoError(tx.AddFiles(s.ctx, []string{filePath}, nil, false))
+
+		committedTbl, err := tx.Commit(s.ctx)
+		s.Require().NoError(err)
+		tbl = committedTbl
+	}
+
+	return tbl, files
+}
+
+func TestRewriteManifests(t *testing.T) {
+	suite.Run(t, &ManifestOperationsTestSuite{})
+}
+
+func (s *ManifestOperationsTestSuite) TestRewriteManifests() {
+	s.Run("ConsolidateManifests", func() {
```

Comment:
there's no shared infrastructure, just make this the function and split the other subtests into their own functions